src/runtime/moduleobject.cs

```diff
@@ -341,6 +341,17 @@ protected override void OnSave(InterDomainContext context)
             // Decref twice in tp_clear, equilibrate them.
             Runtime.XIncref(dict);
             Runtime.XIncref(dict);
+            // destroy the cache(s)
+            foreach (var pair in cache)
+            {
+                Runtime.PyDict_DelItemString(dict, pair.Key);
+                pair.Value.DecrRefCount();
+            }
+            // Trying to remove a key that's not in the dictionary may 
+            // raise an error. We don't care about it.
+            Runtime.PyErr_Clear();
```

Comment:
You must check for the specific error type after each call. See `PyObject.GetAttrOrDefault` for example.